hal/mbed_lp_ticker_api.cpp

```diff
@@ -0,0 +1,208 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2015 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "hal/lp_ticker_api.h"
+
+#if DEVICE_LOWPOWERTIMER
+
+#define DELAY_TICKS 3
+
+#if defined(DELAY_TICKS) && (DELAY_TICKS > 0)
+#include "Timeout.h"
+#include "mbed_critical.h"
+static void set_interrupt_wrapper(timestamp_t timestamp);
+#else
+#define set_interrupt_wrapper       lp_ticker_set_interrupt
```

Comment:
should be uppercase

Reply:
Was about to disagree, but actually, I do think this is a bit too sneaky - too easy quickly scanning and searching to assume looking at the `ticker_interface_t` that the `set_interface_wrapper()` function visible in this file is always used as `.set_interrupt`.

Suggest having an obvious `.set_interrupt = SET_INTERRUPT_FN`, and then that is obviously defined as either `set_interrupt_wrapper` or `lp_ticker_set_interrupt`. Or just put an ifdef/else for the two alternatives in the structure definition itself, skipping the define.